src/shared/sam/activation.ts

```diff
@@ -287,6 +287,46 @@ async function createYamlExtensionPrompt(): Promise<void> {
             yamlPromptDisposables
         )
 
+        /**
+         * Prompt the user to install the YAML plugin when AWSTemplateFormatVersion becomes available as a top level key
+         * in the document
+         * @param event An vscode text document change event
+         * @returns nothing
+         */
+        async function promptOnAWSTemplateFormatVersion(event: vscode.TextDocumentChangeEvent): Promise<void> {
+            for (const change of event.contentChanges) {
+                const changedLine = event.document.lineAt(change.range.start.line)
+                if (changedLine.text.includes('AWSTemplateFormatVersion')) {
```

Comment:
Does this risk false-positives? E.g. a comment or string could contain "AWSTemplateFormatVersion".

Also we know the line with be in the top 3-4 lines of the doc, right? So we can skip lines that are below that?

Reply:
It has the small potential to have a false-positive but I think if the user is editing yaml, has the AWS toolkit installed, and has `AWSTemplateFormatVersion` in the document then theres a big chance they are looking for support

Alternatively, I could change `includes` to `startsWith` and then it's guaranteed to be key in the yaml document. I think most people probably put AWSTemplateFormatVersion in the top 4 lines but I don't think it's guaranteed by the spec so we might end up missing the rare few people that don't put it there